src/main/resources/templates/oidcRedirect.html

```diff
@@ -14,120 +14,257 @@
+        document.getElementById("post_button").disabled = false;
+      } else {
+        console.warn('Either State or Nonce is not validated', { state: stateValid, nonce: nonceValid });
+      }
+    }
+
+    const setAutoRedirectUrl = (address) => {
+        redirectUrl = address;
+    };
+
+    const verifyAnswer = (uuid, expected_value, data) => {
+      // Verify the state and nonce
+      if (
+        // verify `data` is an object
+        typeof data === "object" &&
+        // Message ID we sent needs to match response
+        data.message_id === uuid &&
+        // verify data.subject = lti.put/get_data.response
+        data.subject === 'lti.put_data.response'
+        )
+        {
+            console.log(
+                `✅ ${
+                data.key.includes("state") ? "State" : "Nonce"
+                } passed ${data.value}`
+            );
+            return true;
+        }
+        console.log(
+            `✅ ${
+            data.key.includes("state") ? "State" : "Nonce"
+            } failure ${data} response`
+        );
+        return false;
+    }
+
+    // Function to send both state and nonce to LMS
+    const sendStateAndNonce = async () => {
+      // send state message
+      const state_uuid = await ltiPostMessage.create_UUID();
+      await ltiPostMessage.sendMessage({
+        subject: 'lti.put_data',
+        key: `state${stateHash}`,
+        value: stateHash,
+        message_id: state_uuid // an ID that we can reference on the LTI page
+      }, origin).then(function (response) {
+        console.info('State response from LMS', response);
+        stateValid = verifyAnswer(state_uuid,stateHash,response);
+        verifyStateAndNonce();
+      }).catch(function (error) {
+        console.error('Error sending state message to LMS', error);
+      });
+      const nonce_uuid = await ltiPostMessage.create_UUID();
+      // send nonce message
+      await ltiPostMessage.sendMessage({
+        subject: 'lti.put_data',
+        key: `nonce${nonce}`,
+        value: nonce,
+        message_id: nonce_uuid // and ID that we can reference on the LTI page
+      }, origin).then(function (response) {
+        console.info('Nonce response from LMS', response);
+        nonceValid = verifyAnswer(nonce_uuid,nonce,response);
+        verifyStateAndNonce();
+      }).catch(function (error) {
+        console.error('Error sending nonce message to LMS', error);
+      });
+    }
+
+    // send state and nonce
+    sendStateAndNonce();
+    if (stateNonceValid){
+        setAutoRedirectUrl(
+           [[${ oidcEndpointComplete }]]
+        );
+    } else {
+        setAutoRedirectUrl('ltiError.html');
+    }
+
+    // Optional: Cleanup if needed when the page is unloaded
+    window.addEventListener('unload', function () {
+      // or call this method directly when postMessage service isn't needed anymore
+      ltiPostMessage.destroy();
+    });
+    }
+    if ([[${initiation_dto.ltiStorageTarget }]]) {
```

Comment:
Could this if statement be combined with the one below it to eliminate the duplicate code?